### ISSUE TITLE
References

### ISSUE THREAD

**Issue by cassiozen** [cassiozen/ts-sandbox#1]
https://medium.com/@martin_hotell/10-typescript-pro-tips-patterns-with-or-without-react-5799488d6680

**Comment by cassiozen** [comment]
Angular Team Presentation: https://angular-presentation.firebaseapp.com/angular/typescript/intro

Typescript cheat sheet: https://devhints.io/typescript

**Comment by cassiozen** [comment]
For Redux:
https://medium.com/@martin_hotell/improved-redux-type-safety-with-typescript-2-8-2c11a8062575
https://medium.com/hackernoon/finally-the-typescript-redux-hooks-events-blog-you-were-looking-for-c4663d823b01

**Comment by cassiozen** [comment]
But can it really reduce bugs?

Paper: [To type or not to type: quantifying detectable bugs in JavaScript Gao et al.](http://earlbarr.com/publications/typestudy.pdf)
TL;DR: both Flow and TypeScript are pretty good, and conservatively either of them can prevent about 15% of the bugs that end up in committed code.

[38% of bugs at Airbnb could have been prevented by TypeScript](https://www.reddit.com/r/typescript/comments/aofcik/38_of_bugs_at_airbnb_could_have_been_prevented_by/) according to postmortem analysis

**Comment by cassiozen** [comment]
https://www.youtube.com/watch?v=MbZoQlmQaWQ&t=3s
